From 2a0bc8dc6d1b562ed84af8de3b9ce3c7f59b9d00 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Jul 2018 15:50:36 +0300
Subject: [PATCH] board: habana: improve remote power script

Improve the power down case when the board is already powered down.
Currently, it waits for the ssh to timeout which much longer than
a ping with 1 second wait...
This also adds a 1 sec delay when the board is up, but in this case
it is neglectable.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/remote_power b/board/habanalabs/common/remote_power
index 97fbca0..eaf1fee 100755
--- a/board/habanalabs/common/remote_power
+++ b/board/habanalabs/common/remote_power
@@ -67,8 +67,11 @@
 	;;
 off)
 	sysrq=/proc/sysrq-trigger
-	sshpass -p123456 ssh root@$pciehost "echo s > $sysrq && echo u > $sysrq && sleep 3 && echo o > $sysrq"
-	sleep 10
+	ping -c 1 -W 1 $pciehost > /dev/null
+	if [ $? -eq 0 ]; then
+		sshpass -p123456 ssh root@$pciehost "echo s > $sysrq && echo u > $sysrq && sleep 3 && echo o > $sysrq"
+		sleep 10
+	fi
 	$off
 	;;
 *)
